File: algorithms/insert-sort/insert-sort.c
#include <stdio.h>

void yun_insert_sort_int(int arr[], int length) {
  int key, value, i;
  for (key=1; key<length; key++) {
      value = arr[key];
      i = key-1;
      while (i>=0 && arr[i]>value) {
          arr[i+1] = arr[i];
          i--;
      }
      arr[i+1] = value;
  }
}

int main() {
  int arr[] = {9,6,5,4,3};
  int length;
  yun_insert_sort_int(arr, 5);
  for (length = 0; length < 5; length++) {
      printf("%d\n", arr[length]);
  }
  return 0;
}

File: algorithms/insert-sort/insert-sort.go
package main
import fmt "fmt"

func yun_insert_sort_int(arr []int) {
    var key, value, prev_key, length int
    length = len(arr)
    for key=1; key<length; key++ {
        value = arr[key]
        
        for prev_key=key-1; prev_key>=0 && arr[prev_key]>value; prev_key-- {
            arr[prev_key+1] = arr[prev_key]
        }
        
        arr[prev_key+1] = value;
    }
}


func main() {
    arr := [5]int{9,6,3,4,5}
    for_sort := arr[:]
    yun_insert_sort_int(for_sort);
    fmt.Println(arr)
}
